core/templates/pages/exploration-editor-page/modal-templates/state-version-history-modal.component.ts

```diff
@@ -109,8 +109,12 @@ export class StateVersionHistoryModalComponent
 
     this.newState = diffData.newState;
     this.oldState = diffData.oldState;
-    this.newStateName = diffData.newState.name;
-    this.oldStateName = diffData.oldState.name;
+    if (diffData.newState && diffData.newState.name !== null) {
```

Comment:
Ditto as above.

Reply:
Added the explanations in the code. PTAL.